index.js

```diff
@@ -533,7 +534,20 @@ function fastifyView (fastify, opts, next) {
     // Append view extension.
     page = getPage(page, 'liquid')
 
-    engine.renderFile(join(templatesDir, page), data, opts)
+    const { root } = engine.options
+
+    // Don't execute it if root path is included, then serve it directly
+    const isRootIncluded = root
+      .map((path) => normalize(path))
+      .find((path) => page.indexOf(path) !== -1)
+
+    const pagePath = (typeof root.length !== 'undefined' && !isRootIncluded)
+      ? root
+          .map((dir) => join(resolve(dir), page))
+          .find((filepath) => existsSync(filepath)) || join(templatesDir, page)
```

Comment:
The `existsSync` function is blocking the event loop: you can accept any other requests while the process is checking if the file exists or not.

we have two options:

- call an `async` function
- if we can't call an `async` function (not sure why), we should cache any previous check. For example, the `index` template will be called many times from the homepage site.. so it is a waste of resource to check every time